### PR TITLE
added docs and Bash script

### PR DESCRIPTION
**Pull request by salman1993**
- added docs for setting up e2e qa system
-  added a Bash script to start the Gateway server, Flask API and the JavaScript app

### CODE REVIEW

File: docs/e2e-qa-system.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        ps_split = ps.split('\t')
+        candidate_passages.append(ps_split[0])
+​
+    answers_list = model.rerank_candidate_answers(question, candidate_passages)
+    answers = []
+    for score, sent in answers_list:
+        answers.append({'passage': sent, 'score': score})
+​
+    return answers
+```
```
Comment (rosequ) [inline docs/e2e-qa-system.md]: @salman1993 @lintool shouldn't we make this a part of `api.py` code instead of asking the user to change it?

Comment (salman1993) [inline docs/e2e-qa-system.md]: I agree but we need the model which in turn needs copying the sm_model directory which exists in a different repo and has to be copied over.
So... if we make this a part of the api.py code at this moment, the whole thing will break so I documented it. The ideal solution (that I propose) will be to have an interface for the Model or Bridge or whatever and all future models will obey that contract. So all we will need to do is import and switch out the model and we can have a pre-built IdentityModel that does nothing but returns the candidate passages.

Comment (rosequ) [inline docs/e2e-qa-system.md]: Or, we can just import across directories. Then the user wouldn't have to copy anything...

Comment (salman1993) [inline docs/e2e-qa-system.md]: So sth like 'from Castor.sm_model.sm_model_bridge import SMModel'? That will also include exporting Castor as a package (pretty good solution) or putting __init__.py in castorini (ugly!)

Comment (rosequ) [inline docs/e2e-qa-system.md]: Why is `__init__.py` ugly? I have seen standard packages using it... Also, couldn't find anything in PEP8 about it... @tuzhucheng any suggestions?

Comment (rosequ) [inline docs/e2e-qa-system.md]: Why is __init__.py ugly? I have seen standard packages using it... Also, couldn't find anything in PEP8 about it... @tuzhucheng any suggestions?

Comment (salman1993) [inline docs/e2e-qa-system.md]: The current directory format is this:
```
- castorini
  ---> Anserini/src/main/python/api.py
  ---> Castor/sm_model/*  -> contains the model
  ---> Castor/sm_model_bridge.py -> contains the bridge file
```
So if we want to use __init__.py here, imagine how many places we have to use it to directly import that model in Anserini/src/main/python from Castor. Wouldn't it be better to just export it as a package and import that?

Comment (tuzhucheng) [inline docs/e2e-qa-system.md]: Following the suggestion of http://stackoverflow.com/a/14509415/2514889, maybe we should create each project as a discrete installable package? This way we can import another project as a module.

As an example, I made some modifications to an open-source python package called uap-python and hosted it on GitHub: https://github.com/tuzhucheng/uap-python. I can just install it from pip using `pip install -e git+https://github.com/tuzhucheng/uap-python.git#egg=ua-parser`. I can then use this as a module from another Python project.

**Comment by salman1993** [comment]
closing this PR - covered by commit #170

**Pull request closed without merging**